starter-code/src/viking.js

```diff
@@ -1,11 +1,124 @@
 // Soldier
-class Soldier {}
+class Soldier {
+  constructor(health, strength) {
+    this.health = health;
+    this.strength = strength;
+  }
+  attack() {
+    return this.strength;
+  }
+  receiveDamage(damage) {
+    this.health = this.health - damage;
+  }
+}
+// const soldier1 = new Soldier(500, 40);
+// console.log(soldier1)
+// console.log(soldier1.receiveDamage(40))
+// console.log(soldier1)
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+  constructor(name, health, strength) {
+    super(health, strength);
+    this.name = name;
+  }
+  receiveDamage(damage) {
+    super.receiveDamage(damage);
+    if (this.health > 0) {
+      return `${this.name} has received ${damage} points of damage`;
+    } else {
+      return `${this.name} has died in act of combat`;
+    }
+  }
+  battleCry() {
+    return "Odin Owns You All!";
+  }
+}
+// var viking2 = new Viking("Pierre", 30, 60);
+// console.log(viking2);
+// console.log(viking2.receiveDamage(40));
+// console.log(viking2);
 
 // Saxon
-class Saxon {}
-
+class Saxon extends Soldier {
+  constructor(health, strength) {
+    super(health, strength);
+  }
+  receiveDamage(damage) {
+    super.receiveDamage(damage);
+    if (this.health > 0) {
+      return `A Saxon has received ${damage} points of damage`;
+    } else {
+      return `A Saxon has died in combat`;
+    }
+  }
+}
+// console.log("---------")
+// var viking2 = new Saxon(90, 60);
+// console.log(viking2);
+// console.log(viking2.receiveDamage(20));
+// console.log(viking2);
 // War
-class War {}
+class War {
+  constructor() {
+    this.vikingArmy = [];
+    this.saxonArmy = [];
+  }
+  addViking(Viking) {
+    this.vikingArmy.push(Viking);
+  }
+  addSaxon(Saxon) {
+    this.saxonArmy.push(Saxon);
+  }
+  vikingAttack() {
+    var randomSaxon = Math.floor(Math.random() * this.saxonArmy.length);
+    var chosenSaxon = this.saxonArmy[randomSaxon];
+    var randomViking = Math.floor(Math.random() * this.vikingArmy.length);
+    var chosenViking = this.vikingArmy[randomViking];
+
+    chosenSaxon.receiveDamage(chosenViking.strength);
+
+    if (chosenSaxon.health <= 0) {
+      this.saxonArmy.splice(randomSaxon, 1);
+      return `A Saxon has died in combat`;
+    } else {
+      return `A Saxon has received ${chosenViking.strength} points of damage`;
+    }
+  }
+  saxonAttack() {
+    var randomSaxon = Math.floor(Math.random() * this.saxonArmy.length);
+    var chosenSaxon = this.saxonArmy[randomSaxon];
+    var randomViking = Math.floor(Math.random() * this.vikingArmy.length);
+    var chosenViking = this.vikingArmy[randomViking];
+
+    chosenViking.receiveDamage(chosenSaxon.strength);
+    if (chosenViking.health <= 0) {
+      this.vikingArmy.splice(randomViking, 1);
+      return `${chosenViking.name} has died in act of combat`;
```

Comment:
To avoid to rewrite the sentence (which is the return of the function receiveDamage() ), you could hold this function into a variable(it will at the same time call the function and save the result) and return this variable at the end.